packages/web3-core/src/web3_context.ts

```diff
@@ -98,9 +104,23 @@ export class Web3Context<
 				this.requestManager,
 				registeredSubscriptions,
 			);
+		} else {
+			this._subscriptionManager = new Web3SubscriptionManager(
+				this.requestManager,
+				Web3Context.getDefaultRegisteredSubscriptions(),
+			);
```

Comment:
If you want to set as default behavior, use it as default value of `registeredSubscriptions`

Reply:
I would not recommend it though, every context should be responsible to register their own set of subscriptions.

Reply:
Thanks. You a right. I will add a setter for `registeredSubscriptions` and pass them from web3_eth package